spec/controllers/projects/contributions_controller_spec.rb

```diff
@@ -110,6 +124,54 @@
     end
   end
 
+  describe 'project' do
+    let(:user){ create(:user).decorate }
+
+    before do
+      allow_any_instance_of(ApplicationController).to receive(:current_user).and_return(user)
+    end
+
+    context 'when the project state is different of online' do
+      let(:project) { create(:project, :failed) }
+
+      it 'should redirect to root path on get new request' do
```

Comment:
I think you can extract that `on get new request` do a context:

```ruby
context 'when the request is a get' do
     it 'should redirect to root path
```

The same for the spec bellow

💅